### PR TITLE
Migrate from TSLint to ESLint (Issue #417)

### PR DESCRIPTION
**Pull request by vinyulis**
# PR Details

Migration from TSLint to ESLinit.

Migration steps:

1) Execute: ng add @angular-eslint/schematics
2) To generate new ESLint files execute: ng g @angular-eslint/schematics:convert-tslint-to-eslint
     Type systelab-components when CLI asks for the project
     Repeat Step 2 again and type showcase when CLI asks for the project
3) Review TSLint rules not migrated. In this case:
    "no-access-missing-member": true,
    "templates-use-public": true,
    "invoke-injectable": true
4) Remove tslint.json and old tslint dependencies: 

npm uninstall tslint --save-dev
npm uninstall codelyzer --save-dev

More information about migration in: 

https://github.com/angular-eslint/angular-eslint#migrating-an-angular-cli-project-from-codelyzer-and-tslint
https://www.bitovi.com/blog/angular-upgrades-painless-migration-from-tslint-to-eslint

Added in .eslintrc.json to lint Typescript code with ESLint for IntelliJ/Webstorm (https://www.jetbrains.com/help/idea/eslint.html#ws_eslint_linting_typescript_files_with_eslint)

npm install @typescript-eslint/parser @typescript-eslint/eslint-plugin --save-dev

{
    "parser": "@typescript-eslint/parser",
    "plugins": [
        "@typescript-eslint"
    ],
    "extends": [
        "plugin:@typescript-eslint/recommended"
    ]
}

ESLint configuration in IntelliJ/Webstorm:

image

This PR does not solve previous TSLint errors o warnings or current ESLint errors or warnings.

## Motivation and Context

See #417 

## How Has This Been Tested

Lint execution: ng lint showcase / ng lint systelab-components
Errors shown by IntelliJ/Webstorm?

## Types of changes

Dependency upgrade: Lint migration

### CODE REVIEW

File: projects/showcase/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{
+  "extends": "../../.eslintrc.json",
```
Comment (oGalceran1) [inline projects/showcase/.eslintrc.json:2]: It extends from this file, but the content is quite similar. Can we clean the shared parts?

**Review by oGalceran1 — DISMISSED** [review]
I have added a comment, but the PR seems OK.

**Review by oGalceran1 — APPROVED** [review]

File: projects/showcase/.eslintrc.json (hunk 1 above)
Comment (vinyulis) [reply projects/showcase/.eslintrc.json:2]: Removed duplicities

**Review by atzurk — APPROVED** [review]
OK with this pull-request, thanks!

**Review by RubKrmon — APPROVED** [review]

**Pull request merged**